### ISSUE TITLE
Podman fails to build properly from scratch

### ISSUE THREAD

**Issue by mascenzi80** [containers/podman#19755]
### Issue Description

I have followed the instructions to install the latest version of Podman from Podman's site, [here](https://podman.io/docs/installation#building-from-scratch). When I get to the point where I need to build podman, I start to see what appears to be multiple failures. 

My best research pointed me to another Bug Report on this site, [here](https://github.com/containers/podman/issues/18680), but the suggestion was to use a new version of Golang, which I am currently having the issue with. 

### Steps to reproduce the issue

Steps to reproduce the issue

I followed the steps at Podmans installation page for building from scratch, [here](https://podman.io/docs/installation#building-from-scratch)
and this article, [here](https://techviewleo.com/how-to-install-podman-on-ubuntu/), which simplified some of the noise around what needed to be done for Ubuntu

I was able to build _Podman_ when I was using Golang version 1.13, which was installed via Aptitude. However, when I attempted to run Podman it failed because it was missing the _Seccomp_ profile. That led me to find suggestions that I need to use version 1.16 or higher, and recently a change that says I should be using at least Version 1.18 of Golang. 

So I'm now installing Golang from file, instead of a package manager. I've downloaded Golang version 1.21, and unpacked it to _/usr/local_ I then added the path to the _/usr/profile_ file, and all users including _root_ are able to run _Go_. 

### Describe the results you received

```
$ _make BUILDTAGS="selinux seccomp" PREFIX=/usr_
fatal: not a git repository (or any of the parent directories): .git
Podman is being compiled without the systemd build tag.         Install libsystemd on Ubuntu or systemd-devel on rpm based    distro for journald support.
CGO_ENABLED=1 GOOS=linux GOARCH=amd64 go build \
         \
        -ldflags ' -X github.com/containers/podman/v4/libpod/define.buildInfo=1692990007 -X github.com/containers/podman/v4/libpod/config._installPrefix=/usr/local -X github.com/containers/podman/v4/libpod/config._etcDir=/etc -X github.com/containers/podman/v4/pkg/systemd/quadlet._binDir=/usr/local/bin -X github.com/containers/common/pkg/config.additionalHelperBinariesDir= ' \
        -tags "selinux seccomp" \
        -o bin/podman ./cmd/podman
fatal: not a git repository (or any of the parent directories): .git
CGO_ENABLED=1 GOOS=linux GOARCH=amd64 go build \
         \
        -ldflags ' -X github.com/containers/podman/v4/libpod/define.buildInfo=1692990013 -X github.com/containers/podman/v4/libpod/config._installPrefix=/usr/local -X github.com/containers/podman/v4/libpod/config._etcDir=/etc -X github.com/containers/podman/v4/pkg/systemd/quadlet._binDir=/usr/local/bin -X github.com/containers/common/pkg/config.additionalHelperBinariesDir= ' \
        -tags "remote exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper containers_image_openpgp" \
        -o bin/podman-remote ./cmd/podman
if [ ! -f bin/podmansh ]; then ln -s podman bin/podmansh; fi
CGO_ENABLED=1 \
        go build \
         \
        -o bin/rootlessport ./cmd/rootlessport
fatal: not a git repository (or any of the parent directories): .git
CGO_ENABLED=1 GOOS=linux GOARCH=amd64 go build \
         \
        -ldflags ' -X github.com/containers/podman/v4/libpod/define.buildInfo=1692990019 -X github.com/containers/podman/v4/libpod/config._installPrefix=/usr/local -X github.com/containers/podman/v4/libpod/config._etcDir=/etc -X github.com/containers/podman/v4/pkg/systemd/quadlet._binDir=/usr/local/bin -X github.com/containers/common/pkg/config.additionalHelperBinariesDir= ' \
        -tags "selinux seccomp" \
        -o bin/quadlet ./cmd/quadlet
hack/markdown-preprocess
if [ ! -x "./test/tools/build/go-md2man" ]; then \
        make -C test/tools build/go-md2man ; \
fi
make[1]: Entering directory '/home/mascenzi/podman-4.6.1/test/tools'
go build -o build/go-md2man ./vendor/github.com/cpuguy83/go-md2man/v2
make[1]: Leaving directory '/home/mascenzi/podman-4.6.1/test/tools'
mkdir -p docs/build/man
```

I also attempted to run the install, but it doesn't even see Golang

`$ _sudo make install PREFIX=/usr_
/usr/bin/bash: go: command not found
/usr/bin/bash: go: command not found
env: ‘go’: No such file or directory
env: ‘go’: No such file or directory
Makefile:175: *** Required variable NATIVE_GOOS value is undefined, whitespace, or empty.  Stop.`

### Describe the results you expected

Ability to run Podman

### podman info output

```yaml
OS: Ubuntu 20.04
Podman Version: 4.6
Golang Version: 1.21
conmon Version: 2.1.2
runc Version 1.1.8
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by flouthoc** [comment]
could you set `GOROOT` and then run the command, also what happens when you run simple `make` without any `BUILDTAG` and `PREFIX`.

**Comment by giuseppe** [comment]
what is the error you get if you just run `make` without any additional flag?

**Comment by mascenzi80** [comment]
sorry for the delay in a response. I just got back from Vacation. 

From within the podman-4.6.1 folder, if I run _make_  alone, I get the same response if I run it as _make BUILDTAGS="selinux seccomp"_ which is the following.

```
~/podman-4.6.1$ make BUILDTAGS="selinux seccomp"
if [ ! -f bin/podmansh ]; then ln -s podman bin/podmansh; fi
if [ ! -x "./test/tools/build/go-md2man" ]; then \
        make -C test/tools build/go-md2man ; \
fi
mkdir -p docs/build/man
```

**Comment by mascenzi80** [comment]
I think I might have figured out what's going on. I guess being away has given me some new insight. 

looking more at the 'go' errors I decided I should move to an unused server and take a look at attempting a second install. So I began the process of attempting to get Podman installed on the second server using this [guide](https://techviewleo.com/how-to-install-podman-on-ubuntu/?expand_article=1). 

It seems I didn't follow this guide before, as I couldn't install Golang using the instructions provided. In the guide it tells me to do the following commands.

```
sudo -i
wget https://storage.googleapis.com/golang/getgo/installer_linux
chmod +x ./installer_linux
./installer_linux
```

however, what ends up happening is I get a 400 Bad Request error

```
root@dcop-dev:~# ./installer_linux
Welcome to the Go installer!
Downloading Go version go1.21.0
time 2023-08-04T20:14:06Z to /root/.go
This may take a bit of time...
Downloading Go from 400 Bad Request failed with HTTP status %!s(MISSING)
root@dcop-dev:~# chmod +x ./installer_linux
```

I remember now that I installed Go, I think via aptitude, and this obviously doesn't give the root access to go.

**Comment by giuseppe** [comment]
looks like you are hitting https://github.com/golang/go/issues/62112

In any case this is not a problem with podman, I am closing the issue, but feel free to comment further.

**Comment by mascenzi80** [comment]
> looks like you are hitting [golang/go#62112](https://github.com/golang/go/issues/62112)
> 
> In any case this is not a problem with Podman, I am closing the issue, but feel free to comment further.

I agree that it's not a problem with Podman directly, but the instruction sets available for installing Podman are not viable and should be updated to reflect the available toolsets. 

[Podman Installation](https://podman.io/docs/installation#building-from-scratch) provides instructions for building from source, but fails to build correctly due to  _getgo_ no longer being maintained. [Issue60951](https://github.com/golang/go/issues/60951)
